BUG: transient iOS CI failure: "Could not parse python long as longdouble: 0 (Invalid argument)"

### Describe the issue:

This has happened at least 5 times in the last 7 days:
* https://github.com/numpy/numpy/actions/runs/29687843516/job/88195238839
* https://github.com/numpy/numpy/actions/runs/29725035175/job/88424784745
* https://github.com/numpy/numpy/actions/runs/29751358583/job/88382403225
* https://github.com/numpy/numpy/actions/runs/30088598613/job/89466494698
* https://github.com/numpy/numpy/actions/runs/30120919119/job/89573083803

@freakboy3742: FYI

### Error message:

From the last run above:

<details>
<summary>Example stack trace</summary>

```
026-07-24T19:48:26.1965080Z  _____ TestMaskedArrayArithmetic.test_comparisons_for_numeric[le-None-g-B] ______
2026-07-24T19:48:26.1965480Z  
2026-07-24T19:48:26.1965790Z  self = <numpy.ma.tests.test_core.TestMaskedArrayArithmetic object at 0x11f729310>
2026-07-24T19:48:26.1966260Z  op = <built-in function le>, dt1 = dtype('uint8'), dt2 = dtype('float64')
2026-07-24T19:48:26.1966530Z  fill = None
2026-07-24T19:48:26.1966650Z  
2026-07-24T19:48:26.1966830Z      @pytest.mark.parametrize('dt1', num_dts, ids=num_ids)
2026-07-24T19:48:26.1967120Z      @pytest.mark.parametrize('dt2', num_dts, ids=num_ids)
2026-07-24T19:48:26.1967400Z      @pytest.mark.parametrize('fill', [None, 1])
2026-07-24T19:48:26.1967630Z      @pytest.mark.parametrize('op',
2026-07-24T19:48:26.1967940Z              [operator.le, operator.lt, operator.ge, operator.gt])
2026-07-24T19:48:26.1969930Z      def test_comparisons_for_numeric(self, op, dt1, dt2, fill):
2026-07-24T19:48:26.1971150Z          # Test the equality of structured arrays
2026-07-24T19:48:26.1976590Z          a = array([0, 1], dtype=dt1, mask=[0, 1], fill_value=fill)
2026-07-24T19:48:26.1977020Z  
2026-07-24T19:48:26.1977160Z          test = op(a, a)
2026-07-24T19:48:26.1977360Z          assert_equal(test.data, op(a._data, a._data))
2026-07-24T19:48:26.1977610Z          assert_equal(test.mask, [False, True])
2026-07-24T19:48:26.1977830Z          assert_(test.fill_value == True)
2026-07-24T19:48:26.1978010Z  
2026-07-24T19:48:26.1978140Z          test = op(a, a[0])
2026-07-24T19:48:26.1978350Z          assert_equal(test.data, op(a._data, a._data[0]))
2026-07-24T19:48:26.1978600Z          assert_equal(test.mask, [False, True])
2026-07-24T19:48:26.1978820Z          assert_(test.fill_value == True)
2026-07-24T19:48:26.1979460Z  
2026-07-24T19:48:26.1979670Z  >       b = array([0, 1], dtype=dt2, mask=[1, 0], fill_value=fill)
2026-07-24T19:48:26.1979950Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-07-24T19:48:26.1980160Z  
2026-07-24T19:48:26.1980340Z  ../app_packages/numpy/ma/tests/test_core.py:1994:
2026-07-24T19:48:26.1980700Z  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2026-07-24T19:48:26.1980990Z  ../app_packages/numpy/ma/core.py:6891: in array
2026-07-24T19:48:26.1981280Z      return MaskedArray(data, mask=mask, dtype=dtype, copy=copy,
2026-07-24T19:48:26.1983300Z  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2026-07-24T19:48:26.1984040Z  
2026-07-24T19:48:26.1985640Z  cls = <class 'numpy.ma.MaskedArray'>, data = [0, 1], mask = [1, 0]
2026-07-24T19:48:26.1986070Z  dtype = dtype('float64'), copy = None, subok = True, ndmin = 0
2026-07-24T19:48:26.1986400Z  fill_value = None, keep_mask = True, hard_mask = False, shrink = True
2026-07-24T19:48:26.1986680Z  order = None
2026-07-24T19:48:26.1986810Z  
2026-07-24T19:48:26.1987020Z      def __new__(cls, data=None, mask=nomask, dtype=None, copy=False,
2026-07-24T19:48:26.1991760Z                  subok=True, ndmin=0, fill_value=None, keep_mask=True,
2026-07-24T19:48:26.1992110Z                  hard_mask=None, shrink=True, order=None):
2026-07-24T19:48:26.1992380Z          """
2026-07-24T19:48:26.1992880Z          Create a new masked array from scratch.
2026-07-24T19:48:26.1993090Z  
2026-07-24T19:48:26.1993200Z          Notes
2026-07-24T19:48:26.1993260Z          -----
2026-07-24T19:48:26.1993400Z          A masked array can also be created by taking a .view(MaskedArray).
2026-07-24T19:48:26.1993450Z  
2026-07-24T19:48:26.1993500Z          """
2026-07-24T19:48:26.1993550Z          # Process data.
2026-07-24T19:48:26.1993630Z          copy = None if not copy else True
2026-07-24T19:48:26.1993720Z  >       _data = np.array(data, dtype=dtype, copy=copy,
2026-07-24T19:48:26.1993820Z                           order=order, subok=True, ndmin=ndmin)
2026-07-24T19:48:26.1994010Z  E       RuntimeError: Could not parse python long as longdouble: 0 (Invalid argument)
2026-07-24T19:48:26.1994060Z  
2026-07-24T19:48:26.1994160Z  ../app_packages/numpy/ma/core.py:2895: RuntimeError
```
</details>

<details>
<summary>List of failing tests</summary>

```
=========================== short test summary info ============================
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-?-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-b-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-h-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-i-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-l-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-B-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-H-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-I-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-L-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-Q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-e-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-f-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-d-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-?]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-b]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-h]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-i]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-l]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-q]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-B]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-H]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-I]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-L]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-Q]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-e]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-f]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-d]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-F]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-g-D]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-F-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[None-D-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-?-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-b-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-h-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-i-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-l-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-B-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-H-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-I-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-L-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-Q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-e-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-f-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-d-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-?]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-b]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-h]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-i]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-l]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-q]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-B]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-H]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-I]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-L]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-Q]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-e]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-f]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-d]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-F]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-g-D]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-F-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_eq_for_numeric[1-D-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-?-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-b-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-h-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-i-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-l-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-B-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-H-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-I-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-L-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-Q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-e-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-f-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-d-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-?]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-b]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-h]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-i]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-l]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-q]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-B]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-H]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-I]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-L]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-Q]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-e]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-f]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-d]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-F]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-g-D]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-F-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[None-D-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-?-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-b-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-h-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-i-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-l-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-B-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-H-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-I-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-L-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-Q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-e-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-f-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-d-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-?]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-b]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-h]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-i]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-l]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-q]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-B]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-H]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-I]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-L]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-Q]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-e]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-f]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-d]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-F]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-g-D]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-F-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_ne_for_numeric[1-D-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-?-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-b-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-h-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-i-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-l-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-B-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-H-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-I-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-L-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-Q-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-e-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-f-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-d-g]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-g-?]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-g-b]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-g-h]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-g-i]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-g-l]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-g-q]
   FAILED ma/tests/test_core.py::TestMaskedArrayArithmetic::test_comparisons_for_numeric[le-None-g-B]
   = 153 failed, 46752 passed, 502 skipped, 2734 deselected, 68 xfailed, 1 xpassed in
```
</details>

### Python and NumPy Versions:

Python 3.13
NumPy main branch

### How does this issue affect you or how did you find it:

Checked recent runs returned by https://github.com/numpy/numpy/actions/workflows/wheels.yml?query=is%3Afailure

--- a/numpy/_core/src/common/numpyos.c
+++ b/numpy/_core/src/common/numpyos.c
@@ -591,7 +591,14 @@ NumPyOS_ascii_strtold(const char *s, char** endptr)
     if (clocale) {
         errno = 0;
         result = strtold_l(s, endptr, clocale);
+        /*
+         * On some platforms (e.g. Darwin/iOS), freelocale can set errno.
+         * We're interested in the strtold_l failure; so save the errno
+         * after that call, and restore after freelocale(). See gh-32124.
+         */
+        int saved_errno = errno;
         freelocale(clocale);
+        errno = saved_errno;
     }
     else {
         if (endptr != NULL) {

--- a/numpy/_core/src/common/numpyos.c
+++ b/numpy/_core/src/common/numpyos.c
@@ -592,7 +592,7 @@ NumPyOS_ascii_strtold(const char *s, char** endptr)
         errno = 0;
         result = strtold_l(s, endptr, clocale);
         /*
-         * On some platforms (e.g. Darwin/iOS), freelocale can set errno.
+         * On some platforms (e.g. Darwin/iOS), successful freelocale can set errno.
          * We're interested in the strtold_l failure; so save the errno
          * after that call, and restore after freelocale(). See gh-32124.
          */

